Heavy Minigun and Dual AC are now not-hardpoint only (and necessary emplaced gun price changes)

maikilangiolo opened the pull request:
## About The Pull Request
1. Dual AC and minigun are now not TAD-only guns (hardpoint locked) but can moved like other emplaced guns
2.1 Prices were changed for emplaced guns based on a different tiers I proposed to Lumi, following a logic of light vs medium vs heavy: MG08 300 -> 400, HSG 800 --> 600, Minigun 750 --> 600 , Dual AC 600 --> 700, TAT 600 --> 800
2.2 HSG, HMG and Minigun are light emplaced guns vs the heavier AGSL and now Dual AC, and the heaviest TAT, which was too cheap to replace and you get one free roundstart anyway.
2.3 Ammo costs adjusted for the HSG. It had a 300 round magazine that costed 100 points vs the TAD hsg that had 500 round for 30 points. I essentially swapped the two around, since paying 300% the cost for 40% less ammo is stupid.
3. Other misc changes to bring the guns in line w/ each other: minigun and HSG (un)deploy time is the same as they are the same class, dual AC deploy time ~~and HP increased~~ to match the AGLS since they are also the same class and intended to be sidegrades. Armor bumped up accoding to #11108 to bring the guns in line with HSG and MG08, since they are no longer safely tucked in tadpole.
4. ~~No other changes have been made to the minigun stats, they fixed minigun is just a marine minigun but fixed~~ removed IFF but added aim mode. Improved RoF to have superior DPS without aim mode, unchanged in aim mode
5. The dual AC was changed into an expensive but chonky gun. 40 damage, slower rof for worse DPS, same pen, same sunder, smaller mag. Hopefully it turned into a chunk chunk chunk gun that is worth its cost

This PR is also intended to let people (probably only me) test these guns and rebalance them if needed for marine use, considering they were freed from req hell. It is likely the dual AC might need this tweak since it looked like it was intended to be a cheap weapon and I bumped it up in cost.
## Why It's Good For The Game
More variety for marines, no more req hell for the beatiful and cool guns Tye made, and hopefully the start of a minor rework or emplaced guns to make them more fun and apply needed changes, both buffs and nerfs (arguably mostly nerfs).
## Changelog
:cl:
add: added the mounted minigun and mounted dual cannon to normal marine use
balance: rebalanced cost of HSG ammo between marine and tadpole version
balance: rebalanced cost of mounted guns according to a sensible hierarchy of power vs availability
/:cl:

Lumipharon commented on code/modules/projectiles/guns/mounted.dm:
```diff
@@ -125,12 +125,25 @@
 
 	max_integrity = 300
-	soft_armor = list(MELEE = 0, BULLET = 100, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, FIRE = 0, ACID = 0)
+	soft_armor = list(MELEE = 0, BULLET = 100, LASER = 0, ENERGY = 0, BOMB = 80, BIO = 100, FIRE = 0, ACID = 20)
```
why does this have giga armour? 100 bullet and 80 bomb?

Lumipharon commented on code/modules/projectiles/ammo_datums.dm:
```diff
@@ -1249,19 +1249,19 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 
 /datum/ammo/bullet/dual_cannon/on_hit_turf(turf/T, obj/projectile/P)
-	P.proj_max_range -= 20
+	P.proj_max_range -= 5
```
this meants you can pierce like 6-7 turfs without issue, which is probably not good.

Lumipharon commented on code/modules/projectiles/guns/mounted.dm:
```diff
@@ -89,7 +89,7 @@
 //MG-2005 mounted minigun
 
-/obj/item/weapon/gun/minigun_nest
+/obj/item/weapon/gun/standard_minigun
```
minor but why is this 'standard' minigun, when there is already a standard minigun?

Lumipharon commented on code/modules/projectiles/guns/mounted.dm:
```diff
@@ -145,24 +158,24 @@
 
 	max_integrity = 300
-	soft_armor = list(MELEE = 0, BULLET = 100, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, FIRE = 0, ACID = 0)
+	soft_armor = list(MELEE = 0, BULLET = 100, LASER = 0, ENERGY = 0, BOMB = 80, BIO = 100, FIRE = 0, ACID = 20)
```
giga armour

Lumipharon commented on code/modules/reqs/supplypacks.dm:
```diff
@@ -182,17 +182,37 @@ WEAPONS
 	name = "HSG-102 mounted heavy smartgun ammo"
 	contains = list(/obj/item/ammo_magazine/tl102)
-	cost = 100
+	cost = 30
```
Why are all these ammo packs so cheap? Literally less than a minion.

FIXTHERADBOXPLEASE commented:
Req prices don't seem to line up with the changes in #11815 no? Having more mounted gun variety will be great though.

maikilangiolo commented:
As far the "standard minigun" thing, I added it desperately thinking that somehow it's what caused my debug to fail, when it was an unrelated missing line elsewhere (since other emplaced guns had the "standard" thing).

Magazine prices for minigun and dual AC are not mine, I forgot to mention that. Tye himself priced them at 3 points, so I also lowered the small HSG mag cost to 3 (now 30)

maikilangiolo commented:
@Lumipharon 
Changes: 
- Dual AC mag 50 --> 80
- dual AC mag size bulky --> normal
- minigun fire delay 0.15 --> 0.10, removed iff, added 0.05 aim mode fire delay. This changes the DPS as Sergei outlines
![Screenshots_2023-01-19-21-44-31](https://user-images.githubusercontent.com/40152611/213556174-a552d092-4015-4385-b43a-b83e6c7aa512.png)
![Screenshots_2023-01-19-21-37-29](https://user-images.githubusercontent.com/40152611/213556180-9fb4e9a9-a5b4-4af5-be99-e22db808f8e0.png)

XANAv2 commented:
Could we potentially keep the old guns on the old vehicle hard points as well as these new ones? As this just makes them exclusive to infantry.

maikilangiolo commented:
> Could we potentially keep the old guns on the old vehicle hard points as well as these new ones? As this just makes them exclusive to infantry.
> 
> Either that or I'm dumb and can't find them.

You can still use them on tad, just not on the hardpoint.

maikilangiolo commented:
nevermind

Lumipharon reviewed (changes requested):
wait, why are random map files showing as being changed? And for WO are those guns the actually deployed versions, or just a gun on the floor someone has to setup?
